plugins/native-script-example/README.textile

```diff
@@ -0,0 +1,83 @@
+
+bc.. .
+|- build.gradle
+|- src
+   |- main
+   |  |- java
+   |  |  |- ... source code ...
+   |- test
+      |- java
+      |  |- ... source code ...
+      |- resources
+         |- ... test resources ...
+
+
+p. An Elasticsearch plugin can be created by following these six steps.
+
+* Create build.gradle file in the root directory of your plugin.
+* Create source code directories:
+** @mkdir -p src/main/java@
+** @mkdir -p src/test/java@
+** @mkdir -p src/test/resources@
+* The gradle plugin @esplugin@ provides all needed tasks to assemble the plugin. It is using the following settings that should be modified to match the project's plugin class name and license file definition.
+
+bc.. esplugin {
+         description 'ElasticSearch Plugin with Native Script Examples.'
+         classname 'org.elasticsearch.examples.nativescript.plugin.NativeScriptExamplesPlugin'
+     }
+
+p. 
+
+* Create main Plugin class in the @src/main/java@ directory. This project is using @org.elasticsearch.examples.nativescript.plugin.NativeScriptExamplesPlugin@ class as an example, so the it has to be saved as @src/main/java/org/elasticsearch/examples/nativescript/plugin/NativeScriptExamplesPlugin.java@
+
+bc.. package org.elasticsearch.examples.nativescript.plugin;
+
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.script.ScriptModule;
+
+public class NativeScriptExamplesPlugin extends Plugin implements ScriptPlugin {
+    /* .... */
+}
+
+p. 
+
+* The plugin can be built using @gradle build@ command and executed within elasticsearch by using @gradle run@.
+
+h2. Adding Native Scripts
+
+p. Now that the plugin infrastructure is complete, it's possible to add a native script.
+
+h3. Is Prime Native Script
+
+p. One of the example scripts in this project is the "is_prime" script that can be used to check if a field contains a possible prime number. The script accepts two parameters @field@ and @certainty@. The @field@ parameter contains the name of the field that needs to be checked and the @certainty@ parameter specifies a measure of the uncertainty that the caller is willing to tolerate. The script returns @true@ if the field contains a probable prime number and @false@ otherwise. The probability that the number for which the script returned @true@ is prime exceeds (1 - 0.5^certainty). The script can be used in "Script Filter":http://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-script-filter.html as well as a "Script Field":http://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-script-fields.html. The implementation of the "is_prime" native script and it's factory can be found in the "IsPrimeSearchScript":https://github.com/elastic/elasticsearch/blob/master/plugins/native-script-example/src/main/java/org/elasticsearch/examples/nativescript/script/IsPrimeSearchScript.java class.
```

Comment:
I think it would be better to have documentation about the examples on the script classes themselves?